keras_nlp/network_tests/test_load_ckpts.py

```diff
@@ -53,3 +53,22 @@ def test_load_bert(self, bert_variant, weights):
     def test_load_vocabularies(self, vocabulary):
         tokenizer = keras_nlp.models.BertPreprocessor(vocabulary=vocabulary)
         tokenizer("The quick brown fox.")
+
+
+@pytest.mark.slow
+class Gpt2CkptTest(tf.test.TestCase, parameterized.TestCase):
+    @parameterized.named_parameters(
```

Comment:
Follow (new) pattern set in #361: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/network_tests/test_load_ckpts.py#L46

Reply:
@jbischof, I sort of liked the parameterized solution, to be honest. The test name is `test_gpt2_{variant}.` instead of just `test_gpt2`. So, we know which variant is currently being tested and which variant failed the test. Isn't the parameterized way, more readable?

Reply:
@mattdangerw and I thought the loop would be more maintainable than a parameterized test. It would be very easy for someone to add a checkpoint and not think to update `network_tests/`.

Reply:
Cool, gotcha! We can also try something like this:

```
@pytest.mark.slow
class Gpt2CkptTest(tf.test.TestCase, parameterized.TestCase):
    @parameterized.named_parameters(
        (
            checkpoint,
            checkpoint,
            gpt2_model_class_by_name(gpt2_checkpoints[checkpoint]["model"]),
        )
        for checkpoint in gpt2_checkpoints
    )
    def test_load(self, checkpoint, gpt2_class):

        model = gpt2_class(weights=checkpoint)
        input_data = {
            "token_ids": tf.random.uniform(
                shape=(1, 1024),
                dtype=tf.int64,
                maxval=model.vocabulary_size,
            ),
            "padding_mask": tf.constant([1] * 1024, shape=(1, 1024)),
        }
        model(input_data)
```

Reply:
Cool idea! If you do this please change the other tests in the file as well